lib/ansible/modules/network/meraki/meraki_rest.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Kevin Breit (@kbreit) <kevin.breit@kevinbreit.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: meraki_rest
+
+short_description: Direct access to the Cisco Meraki management dashboard
+
+version_added: "2.6"
+
+description:
+    - Enables management of Cisco's Meraki line of products through their cloud based dashboard.
+    - More information about the Meraki API can be found at U(https://dashboard.meraki.com/api_docs).
+
+options:
+    auth_key:
+        description:
+            - Authentication key provided by the dashboard. Required if environmental variable MERAKI_KEY is not set.
+        required: no
+    host:
+        description:
+            - FQDN or IP address of Meraki dashboard.
+        required: yes
+    method:
+        description:
+            - The HTTP method of the request.
+            - Using C(delete) is typically used for deleting objects.
+            - Using C(get) is typically used for querying objects.
+            - Using C(put) is typically used for modifying objects.
```

Comment:
So I am interested to know the difference between PUT and POST. And highlight that here.

Reply:
For the sake of discussion, https://stackoverflow.com/questions/107390/whats-the-difference-between-a-post-and-a-put-http-request#107450 has a nice description. There are some subtle differences, but interestingly enough, one of which is idempotency (PUT is idempotent). I'm not sure if it makes sense to explain the differences in the documentation since that's pretty in the weeds. Meraki's documentation does explain whether `POST` or `PUT` should be used for each action. When I do more abstracted modules, I'll need to work out which to use internally, largely by handling idempotency manually.